include/delphyne/roads/road_network.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <memory>
+
+#include <maliput/api/road_network.h>
+
+namespace delphyne {
+namespace roads {
+
+/// Wrapper around a maliput::api::RoadNetwork.
+/// This implementation is needed for the python bindings due to a lack of support when handling smart pointers.
+/// For more information see:
+/// - https://github.com/maliput/delphyne/issues/846
+/// - https://github.com/maliput/maliput_infrastructure/issues/225
+///
+/// TODO(https://github.com/maliput/delphyne/issues/847): Remove this wrapper once the above issues are resolved.
+class RoadNetwork {
```

Comment:
What if we call it RoadNetworkWrapper? Wouldn't it be more accurate?

Also, by looking at the numerous changes in code with road_network->get()->something I suggest we create the following methods in this class (see https://en.cppreference.com/w/cpp/memory/unique_ptr/operator* ):

```C++
maliput::api::RoadNetwork* operator->() const noexcept {
  return rn_.get();
}
```

Reply:
We should also add tests to this class.

Reply:
Done.